src/outputs/harbor_output.ml

```diff
@@ -551,13 +536,16 @@ module Make (T : T) = struct
                       (fun () ->
                         match c.state with
                         | Hello ->
-                            let bdlen = Strings.length burst_data in
-                            c.buffer <- Strings.append c.buffer (Strings.sub burst_data (max 0 (bdlen - burst)) (min bdlen burst));
+                            let bdlen = Strings.Mutable.length burst_data in
+                            let data =
+                              Strings.Mutable.sub burst_data (max 0 (bdlen - burst)) (min bdlen burst)
```

Comment:
You don't need to `sub` here if you do the above change from `keep` to `sub`.

Reply:
I'm not sure what you are referring to. It actually looks to me like that whole `sub` call is not needed now that I'm looking at it. The original code wasn't great (or else, wasn't abstracted like we just did) so its real intent was hidden. I believe the `add`+`keep` is all it was supposed to do, passing `burst_data` as a whole to the client upon connecting.

Reply:
Yes, you already `sub` above, so you don't need it at all here. In my old code `keep` would keep *at least* the required amount of data (we did not have string views) so that I would need to `sub` again when using it.